p2p/libp2p/netMessenger.go

```diff
@@ -229,6 +231,21 @@ func createPubSub(ctxProvider *Libp2pContext, withSigning bool) (*pubsub.PubSub,
 	return ps, nil
 }
 
+func (netMes *networkMessenger) createConnectionMonitor() error {
```

Comment:
Just a thought: the netMessenger with different ConnectionMonitors will be hard to test, because both the factory and the actual connection monitor are created inside the netMessenger.

Reply:
I know, but they need to be kept in line with kad dht discoverer and sharder implementation. This is indeed hard to test as we need to construct all those inner netMessenger subcomponents. Network package resembles more like an integration tests package because of this. Have some ideas to separate the glue code from the logic wrapper in order to improve the code coverage by "pure" unit tests. Will be done at some point.

Reply:
ok